codelab.md

```diff
@@ -0,0 +1,365 @@
+# Codelab
+
+Throughout this codelab, you'll create a [Tiled tree](https://research.swtch.com/tlog#tiling_a_log).
+
+The Tiled tree will be stored on disk using the layout described in the [layout
+directory](api/layout/README.md). Its checkpoint uses the [checkpoint format](https://github.com/transparency-dev/formats/blob/main/log/README.md#checkpoint-format).
+
+## Prelimiary setup
+
+The command-line tools we'll use from this repository can generate tile based logs from leaf
+data stored on your file system. Each file will correspond to a single leaf in
+the tree.
+
+Before we start, let's define a few environment variables:
+
+```bash
+export DATA_DIR="/tmp/myfiles"  # where we'll store input data for the tree
+export LOG_DIR="/tmp/mylog"  # where the tree will be stored
+export LOG_ORIGIN="My Log"  # the origin of the log used by the Checkpoint format
+```
+
+Checkpoints of the log will be signed, and we need a public/private key pair for this.
+
+Use the `generate_keys` command with `--key_name`, a name
+for the signing entity. You can output the public and private keys to files using
+`--out_pub` path and filename for the public key,
+`--out_priv` path and filename for the private key
```

Comment:
This is quite the sentence. It looks like you tried to make it bullet points from the md file. Maybe do that so that it reads cleaner in the rendered file?